src/openMVG/sfm/pipelines/sequential/sequential_SfM.cpp

```diff
@@ -1416,7 +1414,12 @@ bool SequentialSfMReconstructionEngine::Resection(const std::size_t viewIndex)
         const Vec2 xJ = _features_provider->feats_per_view.at(J)[track.at(J)].coords().cast<double>();
 
         // test if the track already exists in 3D
-        if (_sfm_data.structure.count(trackId) != 0)
+        bool trackIdExists;
+#pragma omp critical
```

Comment:
maybe something like that (in pseudocode...)
https://gist.github.com/simogasp/9f1aee568ca3ddc3d0da267992e7e1b0